### PR TITLE
Decode Shield Events in Verifier

### PR DESCRIPTION
**Pull request by patnir**
## Summary

Decodes the shield event when processing the evm transaction
Slight tweaks to the ironfish.sol contract 

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and label it with `breaking-change-rpc` or `breaking-change-sdk`.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish/src/evm/evm.ts [hunk 1]
```diff
@@ -33,17 +35,41 @@ export class IronfishEvm {
     const result = await this.runTx(opts)
 
     // TODO(jwp) from custom opcodes populate shields and unshields
+
+    const shields = []
+
+    for (const log of result.receipt.logs) {
+      if (Buffer.from(log[0]).toString('hex') !== 'globalContractAddress') {
```
Comment (hughy) [inline ironfish/src/evm/evm.ts:42]: is `'globalContractAddress'` a placeholder here until we determine the hex address?

Comment (patnir) [reply ironfish/src/evm/evm.ts:42]: yeah

File: ironfish-contracts/contracts/Ironfish.sol [hunk 2]
```diff
@@ -2,11 +2,12 @@
 pragma solidity ^0.8.18;
 
 contract Ironfish {
```
Comment (jowparks) [inline ironfish-contracts/contracts/Ironfish.sol:4]: Can we move this to the sdk? If we do this we will be able to export and use directly in the rest of the sdk and cli

Comment (patnir) [reply ironfish-contracts/contracts/Ironfish.sol:4]: Yeah I can move this to a separate PR

File: ironfish/src/evm/evm.ts [hunk 3]
```diff
@@ -33,17 +35,42 @@ export class IronfishEvm {
     const result = await this.runTx(opts)
 
     // TODO(jwp) from custom opcodes populate shields and unshields
+
```
Comment (jowparks) [inline ironfish/src/evm/evm.ts:38]: Can we add a test for shields here? One where shield comes from non-globaladdress and one that does? We will probably need to define a global address for this.

**Review by jowparks — APPROVED** [review]

**Review by danield9tqh — APPROVED** [review]

**Pull request merged**